File: Java/src/jakopec/t4polimorfizam/RadniNalog.java
package jakopec.t4polimorfizam;

public class RadniNalog extends Dokument {

    private Osoba radnik;

    @Override
    public void obradiDokument() {
        System.out.println("Obrađujem radni nalog");
    }

    @Override
    public String getOpis() {
        return null;
    }

    public RadniNalog(){

    }

    public RadniNalog(Osoba radnik) {
        this.radnik = radnik;
    }



    public Osoba getRadnik() {
        return radnik;
    }

    public void setRadnik(Osoba radnik) {
        this.radnik = radnik;
    }
}

File: Kotlin/src/main/kotlin/t4polimorfizam/RadniNalog.kt
package t4polimorfizam

/**
 * @author Matija Skrletović
 **/

class RadniNalog(sifra:Int, brojDokumenta: String, iznos:Double, radnik: Osoba? = null) : Dokument(sifra, brojDokumenta, iznos) {

    override fun obradiDokument() = println("Obrađujem radni nalog")
    override fun getOpis(): String? = null

}